### DISCUSSION TITLE
how to installed lammps with deepmd-kit

### DISCUSSION THREAD

**Discussion by zjd11** [deepmodeling/deepmd-kit#642] [Q&A]
I have installed deepmd-kit by sh,but if found lmp can not be paralleled through MPI, only OpenMP can be used, so I copy the User-deepmd from the share folder to the SRC folder in lammps, and use make yes-user-deepmd, make User-deepmd to install, but there is an error. How to solve it?
the error is as follows:
make[1]: Entering directory '/public/home/nsccsz_user131/bin/lammps-29Oct20/src/Obj_user-deepmd'
mpicxx -g -O3  -DLAMMPS_GZIP -DLAMMPS_MEMALIGN=64   -std=c++11 -DHIGH_PREC   -I/public/home/nsccsz_user131/bin/deepmd-kit133/include -I/public/home/nsccsz_user131/bin/deepmd-kit133/include -I/public/home/nsccsz_user131/bin/deepmd-kit133/include/deepmd  -DMPICH_SKIP_MPICXX -DOMPI_SKIP_MPICXX=1    -c ../main.cpp
/public/home/nsccsz_user131/bin/deepmd-kit133/bin/mpicxx: line 299: x86_64-conda_cos6-linux-gnu-c++: command not found
make[1]: *** [Makefile:109: main.o] Error 127
make[1]: Leaving directory '/public/home/nsccsz_user131/bin/lammps-29Oct20/src/Obj_user-deepmd'
make: *** [Makefile:212: user-deepmd] Error 2

**Comment by njzjz** [comment]
`lmp` was built with MPI. You can use `lmp -h` to see the MPI information.

For this error, you may consider to remove `/public/home/nsccsz_user131/bin/deepmd-kit133/bin` out of your `PATH`.

**Reply by zjd11** [reply]
Thanks,
I have removed the deepmd-kit out of path,but new errors are coming,
In file included from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/tensor.pb.h(35),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/attr_value.pb.h(38),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/function.pb.h(38),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/graph.pb.h(35),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/public/session.h(23),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/deepmd/common.h(3),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/deepmd/NNPInter.h(3),
                 from ../pair_nnp.h(24),
                 from ../fix_dplr.h(12),
                 from ../fix_dplr.cpp(13):
/public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/resource_handle.pb.h(332): error: nonvirtual function cannot be declared with "final" modifier
    void SerializeWithCachedSizes(
         ^

In file included from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/tensor.pb.h(35),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/attr_value.pb.h(38),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/function.pb.h(38),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/graph.pb.h(35),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/public/session.h(23),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/deepmd/common.h(3),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/deepmd/NNPInter.h(3),
                 from ../pair_nnp.h(24),
                 from ../fix_dplr.h(12),
                 from ../fix_dplr.cpp(13):
/public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/resource_handle.pb.h(334): error: nonvirtual function cannot be declared with "final" modifier
    ::PROTOBUF_NAMESPACE_ID::uint8* InternalSerializeWithCachedSizesToArray(
                                    ^

In file included from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/tensor.pb.h(35),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/attr_value.pb.h(38),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/function.pb.h(38),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/graph.pb.h(35),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/public/session.h(23),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/deepmd/common.h(3),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/deepmd/NNPInter.h(3),
                 from ../pair_nnp.h(24),
                 from ../fix_dplr.h(12),
                 from ../fix_dplr.cpp(13):
/public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/resource_handle.pb.h(484): error: namespace "google::protobuf::internal" has no member "InternalMetadataWithArena"
    ::PROTOBUF_NAMESPACE_ID::internal::InternalMetadataWithArena _internal_metadata_;
                                       ^

In file included from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/tensor.pb.h(35),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/attr_value.pb.h(38),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/function.pb.h(38),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/graph.pb.h(35),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/public/session.h(23),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/deepmd/common.h(3),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/deepmd/NNPInter.h(3),
                 from ../pair_nnp.h(24),
                 from ../fix_dplr.h(12),
                 from ../fix_dplr.cpp(13):
/public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/resource_handle.pb.h(584): error #140: too many arguments in function call
    device_.ClearToEmpty(&::PROTOBUF_NAMESPACE_ID::internal::GetEmptyStringAlreadyInited(), GetArenaNoVirtual());
                         ^

In file included from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/tensor.pb.h(35),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/attr_value.pb.h(38),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/function.pb.h(38),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/graph.pb.h(35),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/public/session.h(23),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/deepmd/common.h(3),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/deepmd/NNPInter.h(3),
                 from ../pair_nnp.h(24),
                 from ../fix_dplr.h(12),
                 from ../fix_dplr.cpp(13):
/public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/resource_handle.pb.h(618): error: no instance of overloaded function "google::protobuf::internal::ArenaStringPtr::Mutable" matches the argument list
            argument types are: (const std::string *, google::protobuf::Arena *)
            object type is: google::protobuf::internal::ArenaStringPtr
    return device_.Mutable(&::PROTOBUF_NAMESPACE_ID::internal::GetEmptyStringAlreadyInited(), GetArenaNoVirtual());
                   ^
/public/home/nsccsz_user75/bin/deepmd-kit/include/google/protobuf/arenastring.h(206): note: this candidate was rejected because arguments do not match
    std::string* Mutable(const LazyString& default_value, ::google::protobuf::Arena* arena);
                 ^
/public/home/nsccsz_user75/bin/deepmd-kit/include/google/protobuf/arenastring.h(204): note: this candidate was rejected because arguments do not match
    std::string* Mutable(EmptyDefault, ::google::protobuf::Arena* arena);
                 ^

In file included from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/tensor.pb.h(35),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/attr_value.pb.h(38),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/function.pb.h(38),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/graph.pb.h(35),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/public/session.h(23),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/deepmd/common.h(3),
                 from /public/home/nsccsz_user75/bin/deepmd-kit/include/deepmd/NNPInter.h(3),
                 from ../pair_nnp.h(24),
                 from ../fix_dplr.h(12),
                 from ../fix_dplr.cpp(13):
/public/home/nsccsz_user75/bin/deepmd-kit/include/tensorflow/core/framework/resource_handle.pb.h(639): error: class "google::protobuf::internal::ArenaStringPtr" has no member "UnsafeArenaRelease"
    return device_.UnsafeArenaRelease(&::PROTOBUF_NAMESPACE_ID::internal::GetEmptyStringAlreadyInited(),
                   ^

compilation aborted for ../fix_dplr.cpp (code 4)
make[1]: *** [Makefile:109: fix_dplr.o] Error 4
make[1]: Leaving directory '/public/home/nsccsz_user75/bin/lammps-29Oct20/src/Obj_mpi'
make: *** [Makefile:212: mpi] Error 2'

What should I do？

**Reply by njzjz** [reply]
This is due to conflict header files of packages in the installer. Thus, it is not recommended to build with that environment. You may create a new environment and install `libdeepmd` alone.
